### PR TITLE
Psalm integration

### LINKED ISSUE #5: Psalm integration
### Feature Request

|    Q        |   A
|------------ | ------
| QA    | yes

#### Summary

As decided during the [Technical-Steering-Committee Meeting on August 3rd, 2020](https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-08-03-TSC-Minutes.md#vote-on-static-analysis-tooling), Laminas wants to implement [vimeo/psalm](https://github.com/vimeo/psalm) in all packages.

Implementing psalm is quite easy.

#### Required
- [ ] Create a `psalm.xml` in the project root
- [ ] Copy and paste the contents from this [psalm.xml.dist](https://github.com/mezzio/repo-template/blob/6124090610ef1c6d6b0c80a063cafacc3ec68f38/psalm.xml.dist)
- [ ] Run `$ composer require --dev vimeo/psalm`
- [ ] Run `$ vendor/bin/psalm --set-baseline=psalm-baseline.xml`
- [ ] Add a composer script `static-analysis` with the command `psalm --shepherd --stats`
- [ ] Add a new line to `script:` in `.travis.yml`: `- if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi`
- [ ] Remove phpstan from the project (`phpstan.neon.dist`, `.travis.yml` entry, `composer.json` `require-dev` and `scripts`)

##### Optional
- [ ] Fix as many psalm errors as possible.

### PR DESCRIPTION
**Pull request by jslmorrison**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | /no
| New Feature   | yes/
| RFC           | no
| QA            | yes

### Description

Add vimeo/psalm integration.

### CODE REVIEW

**Review by Ocramius — CHANGES_REQUESTED** [review]
Baseline needs expanding 🤔

File: psalm.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<psalm
+    totallyTyped="true"
```
Comment (Ocramius) [inline psalm.xml]: We should probably use `errorLevel="1"` instead

Comment (jslmorrison) [reply psalm.xml]: Updated that parameter now as suggested.

**Comment by Ocramius** [comment]
See CI failures: the generated baseline is missing most reported errors 🤔

**Comment by jslmorrison** [comment]
> See CI failures: the generated baseline is missing most reported errors

Hi - I don't see these errors when running static-analysis locally. Do you want them suppressed by expanding the baseline in the following manner?

```
<PropertyNotSetInConstructor>
    <errorLevel type="suppress">
        <referencedProperty name="Mezzio\Migration\MigrateCommand::$input" />
    </errorLevel>
</PropertyNotSetInConstructor>
```

**Comment by Ocramius** [comment]
No, they should just appear as "known issues" in the baseline 🤔

Check what CI did run here: possibly some differences between local setup and this?

File: psalm.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    <projectFiles>
+        <directory name="bin"/>
+        <directory name="src"/>
+        <directory name="test"/>
+        <ignoreFiles>
+            <directory name="vendor"/>
+        </ignoreFiles>
+        <ignoreFiles>
+            <directory name="vendor" />
+        </ignoreFiles>
```
Comment (Ocramius) [inline psalm.xml:16]: Duplicate? 🤔

**Review by Ocramius — APPROVED** [review]
Besides the duplication, this looks much better now, thanks @jslmorrison!

**Review by Ocramius — APPROVED** [review]
Thanks @jslmorrison!

**Pull request merged**